File: kbe/src/client/unity3d/ScriptModule.cs
namespace KBEngine
{
  	using UnityEngine; 
	using System; 
	using System.Collections; 
	using System.Collections.Generic;
	
    public class ScriptModule
    {
		public string name;
		
		public Dictionary<string, Property> propertys = new Dictionary<string, Property>();
		public Dictionary<UInt16, Property> idpropertys = new Dictionary<UInt16, Property>();
		
		public Dictionary<string, Method> methods = new Dictionary<string, Method>();
		public Dictionary<string, Method> base_methods = new Dictionary<string, Method>();
		public Dictionary<string, Method> cell_methods = new Dictionary<string, Method>();
		
		public Dictionary<UInt16, Method> idmethods = new Dictionary<UInt16, Method>();
		public Dictionary<UInt16, Method> idbase_methods = new Dictionary<UInt16, Method>();
		public Dictionary<UInt16, Method> idcell_methods = new Dictionary<UInt16, Method>();
		
		public Type script = null;
		
		public ScriptModule(string modulename)
		{
			name = modulename;
			script = Type.GetType("KBEngine." + modulename);
		}
    }
    
} 

File: kbe/res/client/sdk_templates/ue4/Source/KBEnginePlugins/Private/ScriptModule.cpp

#include "ScriptModule.h"
#include "Method.h"
#include "Property.h"
#include "Entity.h"
#include "KBDebug.h"

ScriptModule::ScriptModule(const FString& moduleName):
	name(moduleName),
	usePropertyDescrAlias(false),
	useMethodDescrAlias(false),
	propertys(),
	idpropertys(),
	methods(),
	base_methods(),
	cell_methods(),
	idmethods(),
	idbase_methods(),
	idcell_methods(),
	pEntityCreator(NULL)
{
	pEntityCreator = EntityFactory::findCreator(moduleName);
	if (!pEntityCreator)
		SCREEN_ERROR_MSG("ScriptModule::ScriptModule(): can't load scriptSodule(KBEngine.%s)!", *moduleName);
}

ScriptModule::~ScriptModule()
{
}
